server.py

```diff
@@ -85,20 +95,26 @@ def file_exists():
         return 'File not found', 404
 
 # Helper method for sending a file to another server
-def move_file(request):
+def move_file(request, method, ip_address):
     metadata = getattr(g, 'metadata', None)
     file_uuid = request.args.get('uuid')
+    file_path = UPLOADED_FOLDER + '/' + file_uuid
+    if not os.path.exists(file_path):
+        return make_response('File not found', 404)
     destination = request.args.get('destination')
     destination_with_endpoint = destination + '/write'
-    write_request = util.construct_post_request(destination_with_endpoint, file_uuid)
+    files = {'file': open(file_path, 'rb')}
+    write_request = requests.post(url, files)
     metadata.update_file_stored(file_uuid, destination)
+    logger.log(filename, ip_address, app.config['HOST'], method, write_request.status_code, os.path.getsize(file_path))
     return write_request
 
 # Transfers the file. This API call should not be open to all users.
 @app.route('/transfer', methods=['PUT'])
 def transfer():
+    ip_address = request.remote_addr if request.args.get('ip') is None else request.args.get('ip')
     metadata = getattr(g, 'metadata', None)
-    write_request = move_file(request)
+    write_request = move_file(request, 'TRANSFER', ip_address)
     if write_request.status_code == 201:
         os.remove(os.path.join(app.config['UPLOAD_FOLDER'], file_uuid))
         metadata.delete_file_stored(request.args.get('uuid'), request.args.get('destination'))
```

Comment:
hm. think i put this on the last pull request too, but shouldn't the second argument to this be the `app.config['HOST']`?